### PR TITLE
[master][KOGITO-1573] Optional full testing for Operator PRs

### PR DESCRIPTION
**Pull request by jakubschwan**
Jira ticket: https://issues.redhat.com/browse/KOGITO-1573
Description: For some cases it would be good to have option to run all tests for PR. This should be optional.

Create pipeline for full test execution in PR.

Signed-off-by: Jakub Schwan <jschwan@redhat.com>

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: Jenkinsfile.pr-full-test [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                }
+            }
+        }
+        stage('Initialize') {
+            steps {
+                sh "mkdir -p ${WORKING_DIR} && cd ${WORKSPACE} && cp -Rap * ${WORKING_DIR}"
+                sh "set +x && oc login --token=\$(oc whoami -t) --server=${OPENSHIFT_API} --insecure-skip-tls-verify"
+            }
+        }
+        stage('Check if Koigto Operator image exists') {
```
Comment (radtriste) [inline Jenkinsfile.pr-full-test:28]: small typo `Koigto`

File: Jenkinsfile.pr-full-test [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                expression {
+                    return operatorImageNotExistsInOpenshift
+                }
+            }
+            steps {
+                dir ("${WORKING_DIR}") {
+                    sh """
+                        set +x && podman login -u jenkins -p \$(oc whoami -t) --tls-verify=false ${OPENSHIFT_REGISTRY}
+                        cd version/ && TAG_OPERATOR=\$(grep -m 1 'Version =' version.go) && TAG_OPERATOR=\$(echo \${TAG_OPERATOR#*=} | tr -d '"')
+                        podman tag quay.io/kiegroup/kogito-cloud-operator:\${TAG_OPERATOR} ${OPENSHIFT_REGISTRY}/openshift/kogito-cloud-operator:pr-\$(echo \${GIT_COMMIT} | cut -c1-7)
```
Comment (radtriste) [inline Jenkinsfile.pr-full-test:86]: could we put this in environment variable ?
like
`PR_NUMBER=${sh(script:"echo \${TAG_OPERATOR#*=} | tr -d '\"'", returnStdOut: true)}`

or in `Initialize` stage:
`env.PR_NUMBER=sh(script:"echo \${TAG_OPERATOR#*=} | tr -d '\"'", returnStdOut: true)`

Comment (radtriste) [reply Jenkinsfile.pr-full-test:86]: could also be changed in other Jenkinsfiles ;)

**Comment by jakubschwan** [comment]
run full test to try triggering of the full job

File: Jenkinsfile.pr-full-test (hunk 2 above)
Comment (spolti) [reply Jenkinsfile.pr-full-test:86]: +1

File: Jenkinsfile.pr-full-test [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            when {
+                expression {
+                    return operatorImageNotExistsInOpenshift
+                }
+            }
+            steps {
+                dir ("${WORKING_DIR}") {
+                    sh """
+                        set +x && podman login -u jenkins -p \$(oc whoami -t) --tls-verify=false ${OPENSHIFT_REGISTRY}
+                        cd version/ && TAG_OPERATOR=\$(grep -m 1 'Version =' version.go) && TAG_OPERATOR=\$(echo \${TAG_OPERATOR#*=} | tr -d '"')
```
Comment (spolti) [inline Jenkinsfile.pr-full-test:85]: it could be on a env as well as it is used more than once.

**Comment by jakubschwan** [comment]
try again run full test to try triggering of the full job

File: Jenkinsfile.pr-full-test [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+            steps {
+                dir ("${WORKING_DIR}") {
+                    deleteDir()
+                }
+            }
+        }
+        stage('Initialize') {
+            steps {
+                sh "mkdir -p ${WORKING_DIR} && cd ${WORKSPACE} && cp -Rap * ${WORKING_DIR}"
+                sh "set +x && oc login --token=\$(oc whoami -t) --server=${OPENSHIFT_API} --insecure-skip-tls-verify"
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply Jenkinsfile.pr-full-test:25]: If this is a Jenkins node on OCP, we will have this info mounted in the pod via the Jenkins Service Account.

File: Jenkinsfile.pr-full-test (hunk 2 above)
Comment (jakubschwan) [reply Jenkinsfile.pr-full-test]: yes, good idea. I'll open a new ticket for this topic

**Comment by jakubschwan** [comment]
run full test

**Comment by jakubschwan** [comment]
run full test (now with different user)

**Comment by jakubschwan** [comment]
run full test (now with different user)

**Comment by ricardozanini** [comment]
Should this be on 0.9.x branch as well?

**Comment by jakubschwan** [comment]
run full test

**Comment by jakubschwan** [comment]
run full test

**Comment by jakubschwan** [comment]
run full test

File: Jenkinsfile.pr-full-test (hunk 2 above)
Comment (jakubschwan) [reply Jenkinsfile.pr-full-test]: https://issues.redhat.com/browse/KOGITO-1679

**Comment by ricardozanini** [comment]
The build is failing. Let's wait for a resolution before merging this one. @jakubschwan is this error related to the fact that we haven't merged the PR?

**Comment by radtriste** [comment]
problem seems to come from the pulling from internal Openshift registry. There is still a problem of parallel pulling ...

**Comment by sutaakar** [comment]
Is there a way to see test results without being authenticated in the Jenkins?

**Comment by jakubschwan** [comment]
@sutaakar not right now, I talked about it with @msmagnanijr and he is working on that.

**Comment by ricardozanini** [comment]
> problem seems to come from the pulling from internal Openshift registry. There is still a problem of parallel pulling ...

Well, I'll put this onhold then until we solve this issue.

**Comment by jakubschwan** [comment]
Include changes introduce in #270

**Comment by ricardozanini** [comment]
The Jenkins check is not running any more? Or am I missing something?

**Comment by jakubschwan** [comment]
@ricardozanini yes that's expected, you need to add comment with phrase "run full test" to the PR to trigger the full testing Jenkins job.

**Comment by ricardozanini** [comment]
@jakubschwan but why I'm seeing a failed check now? :(

**Comment by jakubschwan** [comment]
PR updated to latest changes on master. Jenkins run full test

**Comment by spolti** [comment]
still failing =/

**Comment by radtriste** [comment]
Link does not seem correct for the failing.
@jakubschwan Seems the error on the job run, not in the tests that were correctly started
@msmagnanijr Seems the node was stopped during the tests (btw should also kill remaining cucumber projects existing if any)
```
Cannot contact operator-slave-65jrm: java.lang.InterruptedException
Resuming build at Wed May 06 12:04:48 UTC 2020 after Jenkins restart
```

**Comment by msmagnanijr** [comment]
> Link does not seem correct for the failing.
> @jakubschwan Seems the error on the job run, not in the tests that were correctly started
> @msmagnanijr Seems the node was stopped during the tests (btw should also kill remaining cucumber projects existing if any)
> 
> ```
> Cannot contact operator-slave-65jrm: java.lang.InterruptedException
> Resuming build at Wed May 06 12:04:48 UTC 2020 after Jenkins restart
> ```

Yes, you're right. Unfortunately yesterday I had to change the amount of memory / CPU because the cluster was not supporting many executions in parallel, so I had to stop a few times. Today it should work.

**Comment by jakubschwan** [comment]
Looks like a job get lost, I'm going to recreate the job on Jenkins

**Comment by jakubschwan** [comment]
job recreated, trying to run full test

**Comment by jakubschwan** [comment]
run full test

**Comment by jakubschwan** [comment]
run full test

**Comment by jakubschwan** [comment]
run full test

**Comment by ricardozanini** [comment]
@Kaitou786 @radtriste @jakubschwan @sutaakar  is this one still valid?

**Comment by Kaitou786** [comment]
> @Kaitou786 @radtriste @jakubschwan @sutaakar is this one still valid?

This would be nice to have, but given our resource problem on AWS Openshift, it is most likely to fail. We can set it up to run on RHBA Jenkins but that means we would have a problem in exposing the logs. IMO we either put it on hold or close this one until we have a way to expose the logs from RHBA Jenkins.

**Comment by radtriste** [comment]
I would say you can close that one and we will be in touch with @jakubschwan once we move to RHBA Jenkins

**Comment by ricardozanini** [comment]
> I would say you can close that one and we will be in touch with @jakubschwan once we move to RHBA Jenkins

you mean this one?

**Comment by radtriste** [comment]
yeah I mean this PR. The issue should stay opened

**Pull request closed without merging**